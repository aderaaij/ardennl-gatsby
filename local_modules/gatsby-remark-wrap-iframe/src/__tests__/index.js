const Remark = require(`remark`)
const find = require(`unist-util-find`)
const _ = require(`lodash`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

const extractObjectTag = mdast =>
  _.reduce(
    mdast.children[0].children,
    (result, child) => result + child.value,
    ``
  )

const extractIframeTag = mdast => mdast.children[0].value

describe(`gatsby-remark-responsive-iframe`, () => {
  ;[`iframe`, `object`].forEach(tag => {
    it(`transforms an ${tag} with unitless width and height`, async () => {
      const markdownAST = remark.parse(`
<${tag} url="http://www.example.com/" width="600" height="400"></${tag}>
    `)

      const transformed = await plugin({ markdownAST })
      const node = find(transformed, function(node) {
        return node.type === `unknown`
      })
      expect(node).toBeDefined()
      expect(node.data).toBeDefined()
      expect(node.data.hChildren).toBeDefined()
      expect(node.data.hChildren.length).toBeGreaterThan(0)
      const [child] = node.data.hChildren
      expect(child.value).toMatchSnapshot()
    })
  })
  ;[`iframe`, `object`].forEach(tag => {
    it(`transforms an ${tag} with pixel width and height`, async () => {
      const markdownAST = remark.parse(`
<${tag} url="http://www.example.com/" width="600px" height="400px"></${tag}>
    `)

      const transformed = await plugin({ markdownAST })
      const node = find(transformed, function(node) {
        return node.type === `unknown`
      })
      expect(node).toBeDefined()
      expect(node.data).toBeDefined()
      expect(node.data.hChildren).toBeDefined()
      expect(node.data.hChildren.length).toBeGreaterThan(0)
      const [child] = node.data.hChildren
      expect(child.value).toMatchSnapshot()
    })
  })

  const shouldntTransform = [
    [`100%`, `100`],
    [`100`, `100%`],
    [`invalid`, `100`],
    [`100`, `invalid`],
  ]

  _.map(shouldntTransform, ([width, height]) => {
    ;[`iframe`, `object`].forEach(tag => {
      it(`doesn't transform an ${tag} with dimensions: '${width}' '${height}'`, async () => {
        const markdownAST = remark.parse(`
  <${tag} url="http://www.example.com/" width="${width}" height="${height}"></${tag}>
      `)
        const transformed = await plugin({ markdownAST })
        // Note: Remark treats iframes (block level) and object (inline) tags
        // differently, wrapping an object tag in <p></p> tags. It also parses
        // out the object tag into three separate nodes - one for the opening
        // tag, one for the closing tag and one for a newline inside. So for any
        // tests that recieve untransformed node back from the plugin, we strip
        // the p tags and combine the nodes into a single html string.
        if (tag === `iframe`) {
          const iframeHTML = extractIframeTag(transformed)
          expect(iframeHTML).toBeDefined()
          expect(iframeHTML).toMatchSnapshot()
        } else {
          const objectHTML = extractObjectTag(transformed)
          expect(objectHTML).toBeDefined()
          expect(objectHTML).toMatchSnapshot()
        }
      })
    })
  })
})
