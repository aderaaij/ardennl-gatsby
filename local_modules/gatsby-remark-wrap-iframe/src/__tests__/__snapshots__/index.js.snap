// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-responsive-iframe doesn't transform an iframe with dimensions: '100%' '100' 1`] = `
"  <iframe url=\\"http://www.example.com/\\" width=\\"100%\\" height=\\"100\\"></iframe>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an iframe with dimensions: '100' '100%' 1`] = `
"  <iframe url=\\"http://www.example.com/\\" width=\\"100\\" height=\\"100%\\"></iframe>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an iframe with dimensions: '100' 'invalid' 1`] = `
"  <iframe url=\\"http://www.example.com/\\" width=\\"100\\" height=\\"invalid\\"></iframe>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an iframe with dimensions: 'invalid' '100' 1`] = `
"  <iframe url=\\"http://www.example.com/\\" width=\\"invalid\\" height=\\"100\\"></iframe>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an object with dimensions: '100%' '100' 1`] = `
"  <object url=\\"http://www.example.com/\\" width=\\"100%\\" height=\\"100\\"></object>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an object with dimensions: '100' '100%' 1`] = `
"  <object url=\\"http://www.example.com/\\" width=\\"100\\" height=\\"100%\\"></object>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an object with dimensions: '100' 'invalid' 1`] = `
"  <object url=\\"http://www.example.com/\\" width=\\"100\\" height=\\"invalid\\"></object>
      "
`;

exports[`gatsby-remark-responsive-iframe doesn't transform an object with dimensions: 'invalid' '100' 1`] = `
"  <object url=\\"http://www.example.com/\\" width=\\"invalid\\" height=\\"100\\"></object>
      "
`;

exports[`gatsby-remark-responsive-iframe transforms an iframe with pixel width and height 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: NaN%; position: relative; height: 0; overflow: hidden;\\"
          >
            <iframe url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></iframe>
    
          </div>
          "
`;

exports[`gatsby-remark-responsive-iframe transforms an iframe with unitless width and height 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: 66.66666666666666%; position: relative; height: 0; overflow: hidden;\\"
          >
            <iframe url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></iframe>
    
          </div>
          "
`;

exports[`gatsby-remark-responsive-iframe transforms an object with pixel width and height 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: NaN%; position: relative; height: 0; overflow: hidden;\\"
          >
            <object url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></object>
          </div>
          "
`;

exports[`gatsby-remark-responsive-iframe transforms an object with unitless width and height 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: 66.66666666666666%; position: relative; height: 0; overflow: hidden;\\"
          >
            <object url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></object>
          </div>
          "
`;
